### PR TITLE
fix(cli): don't count verbose diagnostics

### PR DESCRIPTION
**Pull request by ematipico**
## Summary

Fixes https://github.com/biomejs/biome/issues/1512

## Test Plan

Added two new test cases

### CODE REVIEW

File: crates/biome_cli/src/execute/traverse.rs [hunk 1]
```diff
@@ -246,10 +247,10 @@ struct DiagnosticsPrinter<'ctx> {
     ///  Execution of the traversal
     execution: &'ctx Execution,
     /// The maximum number of diagnostics the console thread is allowed to print
-    max_diagnostics: u16,
+    max_diagnostics: u64,
```
Comment (Conaclos) [inline crates/biome_cli/src/execute/traverse.rs:250]: Why did you change the type? `u32` could be enough?

**Review by Conaclos — APPROVED** [review]

File: crates/biome_cli/src/execute/traverse.rs (hunk 1 above)
Comment (ematipico) [reply crates/biome_cli/src/execute/traverse.rs:250]: I had a casting issue among numbers
